r/R/util.R

```diff
@@ -45,3 +45,16 @@ is_list_of <- function(object, class) {
 }
 
 empty_named_list <- function() structure(list(), .Names = character(0))
+
+read_compressed_error <- function(e) {
+  e <- as.character(e)
+  alg <- regmatches(e, gregexpr("(?<=\')(.*?)(?=\')", e, perl = TRUE))[[1]]
```

Comment:
Could you post an example of what the message is that you're matching here? I think the regex here might be overly broad and need some refinement

Reply:
yes, the error is 
```
Error: NotImplemented: Support for codec 'lz4' not built
```

Reply:
therefore I keep what's inside ' '

Reply:
OK, we should look for a bit more than that that. Specifically we should check that the error is about a codec of some sort, and only if it is then extract the codec out and follow this path. We want to pass on errors that are not related to codecs not being built. Look at the following:

```
> arrow::read_feather(system.file("v0.7.1.parquet", package = "arrow"))
Error: Invalid: Not a Feather V1 or Arrow IPC file
```

Also, if I'm looking up the regex you have I see some look ahead / look behind assertions, can you explain what those are doing in there?

Reply:
I think this is more readable:

```suggestion
  alg <- sub(".*Support for codec '(.*)'.*", "\\1", e)
```

Reply:
yes, it is